crates/datastore/src/locking_tx_datastore/committed_state.rs

```diff
@@ -643,17 +644,18 @@ impl CommittedState {
 
             if !deletes.is_empty() {
                 let table_name = &*table.get_schema().table_name;
-                // TODO(centril): Pass this along to record truncated tables.
-                let _truncated = table.row_count == 0;
-                tx_data.set_deletes_for_table(table_id, table_name, deletes.into());
+                let truncated = table.row_count == 0;
```

Comment:
Should we rather let caller of `fn delete_rows` to pass `truncated` as param and maybe just `assert` row counts here?(pass it true if table is from `PendingSchemaChanges`).

Reason is: this code considers empty table also as truncated table for generating `TxData`. 
Which may not be the problem on replay due to `get_table_blob_store_or_create` will create it again but still not seems great.

Reply:
From the link, 
> TRUNCATE TABLE removes all rows from a table, but the table structure and its columns, constraints, indexes, and so on, remain. To remove the table definition in addition to its data, use the DROP TABLE statement.

Makes sense.

Reply:
So the problem is, there rows insrted back during insert. So, we can not mark it truncated before checking that. I will push the fix for it.